ompi/mca/mtl/ofi/mtl_ofi_types.h

```diff
@@ -64,75 +80,169 @@ typedef struct mca_mtl_ofi_component_t {
     mca_mtl_base_component_2_0_0_t super;
 } mca_mtl_ofi_component_t;
 
+/*OFI TAG:
+ * Define 3 different OFI tag distributions:
+ * 1) Support FI_REMOTE_CQ_DATA: No need for source rank in the tag
+ * 2) FALLBACK_1: fallback when no FI_REMOTE_CQ_DATA is supported
+ * 3) FALLBACK_2: Alternative fallback when no FI_REMOTE_CQ_DATA is supported
+ *    with more bits for the communicator ID. Select with
+ *    -mca mtl_ofi_alternative_fallback_tag true
+*/
+
+/* Support FI_REMOTE_CQ_DATA, send the source rank in the CQ data (4 Bytes is the minimum)
+ *  01234567 01234567 01234567 0123 4567 01234567 01234567 01234567 01234567
+ *                                 |    |
+ *           context_id            |prot|          message tag
+ */
+#define MTL_OFI_PROTO_BIT_COUNT         (4)
+
+#define MTL_OFI_CID_BIT_COUNT_DATA      (28)
+#define MTL_OFI_TAG_MASK_DATA           (0x00000000FFFFFFFFULL)
+#define MTL_OFI_TAG_BIT_COUNT_DATA      (32)
+#define MTL_OFI_PROTO_MASK_DATA         (0x0000000F00000000ULL)
+#define MTL_OFI_SYNC_SEND_DATA          (0x0000000100000000ULL)
+#define MTL_OFI_SYNC_SEND_ACK_DATA      (0x0000000900000000ULL)
+
+__opal_attribute_always_inline__ static inline uint64_t
```

Comment:
does this *have* to be inlined?  Thinking you're smarter than the inline is generally a path to madness...

Reply:
This function creates the send tag. Please check previous comments where it was asked to move the macros that create tags to inline functions.